src/js/relayclient/relayclient.js

```diff
@@ -37,11 +38,13 @@ function RelayClient(web3, config) {
     // TODO: require sign() or privKey
     this.config = config || {}
     this.web3 = web3
+    this.httpSend = new HttpWrapper(this.web3)
+    this.serverHelper = this.config.serverHelper || new ServerHelper(this.config.minStake || 0, this.config.minDelay || 0, this.httpSend)
 
     this.RelayRecipient = web3.eth.contract(relayRecipientAbi)
     this.RelayHub = web3.eth.contract(relayHubAbi)
+    this.serverHelper.setHub(this.RelayHub)
```

Comment:
you have a type mix: `RelayHub` is a class, but inside the serverHelper, its `relayHub` (instance), where you assume there's an `address`: there is an address, but its a static one (probably the initial deployed one), and you can't trust it. That's why we read the hub from the contract.